syntax = "proto3";
package packet_ss;
import "base_type.proto";

message MsgRegisterSW
{
    int32 peer_type = 1;
    string listen_ip = 2;
    int32 listen_port = 3;
}

message MsgRegisterWS
{
    int32 server_id = 1;
}

message MsgServerInfoWS
{
    message ServerInfo
    {
        int32 peer_type = 1;
        int32 server_id = 2;
        string listen_ip = 3;
        int32 listen_port = 4;
    }
    repeated ServerInfo server_info = 1;
}

message MsgLogin2NodeGN
{
    int32 peer_type = 1;
    int32 server_id = 2;
}


message MsgReqServerListLM
{
    string account_name = 1;
}

message MsgServerListML
{
    int32 result = 1;
    repeated pb_base.ServerProfile servers = 2;
    MsgReqServerListLM post_back = 3;
}

message MsgSyncLoginServerIdML
{
    int32 server_id = 1;
}

message MsgWorldRegAtMasterWM
{
    int32 server_uid = 1;
}

message MsgWorldRegAtMasterAckMW
{
    int32 result = 1;
}

message MsgReqEnterServerLS
{
    string account_name = 1;
    string account_token = 2;
    int32 server_uid = 3;
    int32 login_serverid = 4;
}

message MsgReqEnterServerResultSL
{
    int32 result = 1;
    string account_name = 2;
    string gate_ip = 3;
    string gate_port = 4;
}
